test/jdk/java/net/HttpURLConnection/Response1xxTest.java

```diff
@@ -203,4 +211,17 @@ public void test1xx() throws Exception {
             Assert.assertEquals(body, EXPECTED_RSP_BODY, "Unexpected response body");
         }
     }
+
+    /**
+     * Tests that when a HTTP/1.1 server sends 101 response code, when the client
+     * didn't ask for a connection upgrade, then the request fails with an exception.
+     */
+    @Test
+    public void test101CausesRequestFailure() throws Exception {
+        final URI requestURI = new URI(requestURIBase + "/test/bye");
+        System.out.println("Issuing request to " + requestURI);
+        final HttpURLConnection urlConnection = (HttpURLConnection) requestURI.toURL().openConnection();
+        // we expect the request to fail because the server unexpectedly sends a 101 response
+        Assert.assertThrows(IOException.class, () -> urlConnection.getResponseCode());
```

Comment:
Should we explicitly wait for ProtocolException here, since it's what we're throwing?

Reply:
I initially thought about that, but since the API javadoc doesn't specifically talk about this sub-type of `IOException`, I decided it's perhaps better not to start expecting it. However, if it's OK for the test to rely on this (internal) knowledge then I'll go ahead and change it.

Reply:
Yes - this is not a conformance test, and ProtocolException is within the bounds. It's OK for a non-regression test to expect the specific subtype that the code change is throwing. My concern here is that the test could be passing if some other kind of errors occurred - like e.g. connect exception. Then we wouldn't be testing what we think we're testing.

Reply:
Thank you. I've now updated the test to expect `ProtocolException`. The test continues to pass with this change.